src/frontend/komponenter/Fagsak/Brevmottaker/LeggTilEndreBrevmottakerModal.tsx

```diff
@@ -45,36 +50,64 @@ const ModalKnapperad = styled.div`
     gap: 1rem;
 `;
 
-interface Props {
-    brevmottakerId?: string;
-}
-
-enum AdresseKilde {
-    MANUELL_REGISTRERING = 'MANUELL_REGISTRERING',
-    OPPSLAG_REGISTER = 'OPPSLAG_REGISTER',
-    OPPSLAG_ORGANISASJONSREGISTER = 'OPPSLAG_ORGANISASJONSREGISTER',
-    UDEFINERT = 'UDEFINERT',
-}
-
-const adresseKilder: Record<AdresseKilde, string> = {
-    MANUELL_REGISTRERING: 'Manuell registrering',
-    OPPSLAG_REGISTER: 'Oppslag i register',
-    OPPSLAG_ORGANISASJONSREGISTER: 'Oppslag i organisasjonsregister',
-    UDEFINERT: 'Udefinert',
-};
+const erMottakerBruker = (mottakerType: MottakerType) =>
+    mottakerType === MottakerType.BRUKER_MED_UTENLANDSK_ADRESSE ||
+    mottakerType === MottakerType.DØDSBO;
 
-export const LeggTilEndreBrevmottakerModal: React.FC<Props> = ({ brevmottakerId }: Props) => {
-    const heading = !brevmottakerId ? 'Legg til brevmottaker' : 'Endre brevmottaker';
+export const LeggTilEndreBrevmottakerModal: React.FC = () => {
     const { visBrevmottakerModal, settVisBrevmottakerModal } = useBehandling();
-    const { skjema, nullstillSkjema, valideringErOk, lagreBrevmottakerOgOppdaterState } =
-        useBrevmottaker();
-    const [adresseKilde, settAdresseKilde] = useState<AdresseKilde>(AdresseKilde.UDEFINERT);
+    const {
+        skjema,
+        nullstillSkjema,
+        adresseKilde,
+        settAdresseKilde,
+        valideringErOk,
+        settVisfeilmeldinger,
+        lagreBrevmottakerOgOppdaterState,
+        brevmottakerTilEndring,
+        settBrevmottakerTilEndring,
+        bruker,
+    } = useBrevmottaker();
+
+    const heading = brevmottakerTilEndring ? 'Endre brevmottaker' : 'Legg til brevmottaker';
+    const [mottakerErBruker, settMottakerErBruker] = React.useState(
+        skjema.felter.mottaker.verdi ? erMottakerBruker(skjema.felter.mottaker.verdi) : false
+    );
+
+    React.useEffect(() => {
+        settMottakerErBruker(
+            skjema.felter.mottaker.verdi ? erMottakerBruker(skjema.felter.mottaker.verdi) : false
+        );
+    }, [skjema.felter.mottaker.verdi]);
+
+    React.useEffect(() => {
+        if (mottakerErBruker) {
+            skjema.felter.navn.validerOgSettFelt(bruker.navn);
+            settAdresseKilde(AdresseKilde.MANUELL_REGISTRERING);
+            nullstillManuellAdresseInput();
```

Comment:
Hvorfor nullstiller vi manuell adresseinput her og på linje 90?

Reply:
Utdatert kode (endret til å ligne mer på det som er lagt til for preutfylling av navn i ba-sak), men nullstillingen er forsåvidt videreført, siden eksisterende input sannsynligvis ikke gjaldt bruker, da mottakertypen må ha vært Verge eller Fullmektig. Unngår f.eks at Norge blir hengende igjen som land etter at mottakertypen er endret til `BRUKER_MED_UTENLANSK_ADRESSE`